clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/OAuthBearerSaslServer.java

```diff
@@ -152,10 +154,7 @@ public void dispose() throws SaslException {
         try {
             callbackHandler.handle(new Callback[] {callback});
         } catch (IOException | UnsupportedCallbackException e) {
-            String msg = String.format("%s: %s", INTERNAL_ERROR_ON_SERVER, e.getMessage());
-            if (log.isDebugEnabled())
-                log.debug(msg, e);
-            throw new SaslException(msg);
+            handleInternalError(e);
```

Comment:
Call we call this something other than `internal` error since that doesn't really convey much. Maybe even `handleCallbackError` would be better.